### PR TITLE
tools/codespell: fix new typos and make it fail when typos are found

### PR DESCRIPTION
**Pull request by aabadie**
### Contribution description

New typos were again introduced by #14065 (witout => without) and #15200 (auto-negotation => auto-negotiation), so not so long after #15138 was merged.

This PR is fixing the newly introduced typos and is also adding _circular_ to the list of ignored words (it's detected as a typo locally but according to wikipedia, it seems valid in this context).

The last commit is enforcing the CI to fail if typos are found. I can removed if needed but I think that would enforce most of the typos to be detected immediately.

### Testing procedure

Run `./dist/tools/codespell/check.sh`

- this PR:

```
$ ./dist/tools/codespell/check.sh
```

- on master:

```
$ ./dist/tools/codespell/check.sh
There are typos in the following files:

core/include/clist.h:18: circularly ==> circular
drivers/include/mii.h:67: auto-negotation ==> auto-negotiation
sys/net/gnrc/link_layer/lorawan/include/gnrc_lorawan_internal.h:314: witout ==> without
```

### Issues/PRs references

Fixes typos introduced in #14065 and #15200

### CODE REVIEW

File: dist/tools/codespell/check.sh
Comment (benpicco) [inline dist/tools/codespell/check.sh:59]: I'm still very much afraid that this will cause more hassle with false positives than any wrong spelling ever would

Comment (aabadie) [reply dist/tools/codespell/check.sh:59]: Same here. But #15138 showed that adding false positives to the list of ignored words is not so difficult. In a PR, codespell is also only run on modified files and this will also limit the unrelated typo found (because now we start at a point where all existing typos are fixed).

Comment (fjmolinas) [reply dist/tools/codespell/check.sh:59]: I'm also a bit on the fence here, same argument as @benpicco. I feel this could be helped if maintainers start checking the output of travis more, even if its green and look at the suggestions.

Comment (aabadie) [reply dist/tools/codespell/check.sh:59]: > this could be helped if maintainers start checking the output of travis more

You know that in practice this never happens (for many reasons)

Comment (fjmolinas) [reply dist/tools/codespell/check.sh:59]: @aabadie pointed out offline that only two new typos went through in 20 days, so it doesn't seem to bad.

File: dist/tools/codespell/check.sh
Comment (benpicco) [inline dist/tools/codespell/check.sh:58]:
```suggestion
    printf "%s\n" "${ERRORS}"
    printf "If those are false positives, add them to %s\n" "${RIOTTOOLS}/codespell/ignored_words.txt"
```

**Review by benpicco — APPROVED** [review]
Ok let's give this a try.
But we should revert this if false positives are becoming a nuisance.

**Comment by aabadie** [comment]
I rebased this PR on top of latest master: maybe some new typos will be found on master. We also have to take care of already opened PRs and re-run Murdock for all opened PRs to make sure there are no hidden typos in there.

**Comment by fjmolinas** [comment]
> I rebased this PR on top of latest master: maybe some new typos will be found on master. We also have to take care of already opened PRs and re-run Murdock for all opened PRs to make sure there are no hidden typos in them.

Should we go through all pr-s marked as ready for ci and re-trigger travis?

**Comment by fjmolinas** [comment]
Otherwise seems that its time to hit GO here right?

**Comment by aabadie** [comment]
> Should we go through all pr-s marked as ready for ci and re-trigger travis?

Just have a look in Travis (or Murdock) if there are typo found. In this case, ask for a rebase, the CI will fail and the contributor will have to fix the typos found.

**Comment by fjmolinas** [comment]
@RIOT-OS/maintainers once we merge this PR code spell will make travis fail if new typos are encountered. But this will not apply to PR's that have already passed travis if no new changes are added to that PR. So before merging PR's please check that codespell is not flagging anything.

**Comment by fjmolinas** [comment]
GO

**Pull request merged**
